Intersect.Client/Entities/Player.cs

```diff
@@ -2372,6 +2376,48 @@ public void DrawOverheadInfoOnHover()
             }
         }
 
+        public override bool ShouldDrawHpBar
+        {
+            get
+            {
+                if (LatestMap == default)
+                {
+                    return false;
+                }
+
+                if (!ShouldDraw)
+                {
+                    return false;
+                }
+
+                var health = Vital[(int)Vitals.Health];
+                if (health < 1)
+                {
+                    return false;
+                }
+
+                var maxHealth = MaxVital[(int)Vitals.Health];
+                var shieldSize = GetShieldSize();
+
+                var player = this;
+                bool isMe = player.Id == Globals.Me.Id,
+                    isFriend = !isMe && Globals.Me.IsFriend(player),
+                    isPartyMate = !isMe && Globals.Me.IsInMyParty(player),
+                    isGuildMate = !isMe && Globals.Me.IsGuildMate(player),
+                    friendSettingMatch = isFriend && Globals.Database.FriendOverheadHpBar,
+                    guildSettingMatch = isGuildMate && Globals.Database.GuildMemberOverheadHpBar;
+
+                return (shieldSize > 0 || health != maxHealth) ||
```

Comment:
Since this part is duplicated, after the short-circuit with `ShouldNotDrawHpBar`, do a if base.ShouldDrawHpBar return true